File: infinitystone/infinitystone.py
def on_button_pressed_a():
    global 인피티니스톤
    인피티니스톤 += 합
    basic.show_number(인피티니스톤)
input.on_button_pressed(Button.A, on_button_pressed_a)

def on_gesture_shake():
    global 합
    합 = randint(1, 6)
    if 합 == 1:
        strip.show_color(neopixel.colors(NeoPixelColors.YELLOW))
    elif 합 == 2:
        strip.show_color(neopixel.colors(NeoPixelColors.RED))
    elif 합 == 3:
        strip.show_color(neopixel.colors(NeoPixelColors.ORANGE))
    elif 합 == 4:
        strip.show_color(neopixel.colors(NeoPixelColors.VIOLET))
    elif 합 == 5:
        strip.show_color(neopixel.colors(NeoPixelColors.BLUE))
    elif 합 == 6:
        strip.show_color(neopixel.colors(NeoPixelColors.GREEN))
input.on_gesture(Gesture.SHAKE, on_gesture_shake)

def on_button_pressed_b():
    basic.show_number(딱)
input.on_button_pressed(Button.B, on_button_pressed_b)

딱 = 0
합 = 0
인피티니스톤 = 0
strip: neopixel.Strip = None
strip = neopixel.create(DigitalPin.P2, 24, NeoPixelMode.RGB_RGB)
strip.clear()
strip.set_brightness(80)
strip.show()

def on_forever():
    global 딱
    딱 = pins.analog_read_pin(AnalogPin.P1)
    if 인피티니스톤 == 21 and 딱 > 200:
        strip.show_rainbow(1, 360)
        while 인피티니스톤 == 21:
            strip.rotate(1)
            strip.show()
            basic.pause(100)
basic.forever(on_forever)

File: infinitystone/infinitystone.js
input.onButtonPressed(Button.A, function () {
    인피티니스톤 += 합
    basic.showNumber(인피티니스톤)
})
input.onGesture(Gesture.Shake, function () {
    합 = randint(1, 6)
    if (합 == 1) {
        strip.showColor(neopixel.colors(NeoPixelColors.Yellow))
    } else if (합 == 2) {
        strip.showColor(neopixel.colors(NeoPixelColors.Red))
    } else if (합 == 3) {
        strip.showColor(neopixel.colors(NeoPixelColors.Orange))
    } else if (합 == 4) {
        strip.showColor(neopixel.colors(NeoPixelColors.Violet))
    } else if (합 == 5) {
        strip.showColor(neopixel.colors(NeoPixelColors.Blue))
    } else if (합 == 6) {
        strip.showColor(neopixel.colors(NeoPixelColors.Green))
    }
})
input.onButtonPressed(Button.B, function () {
    basic.showNumber(딱)
})
let 딱 = 0
let 합 = 0
let 인피티니스톤 = 0
let strip: neopixel.Strip = null
strip = neopixel.create(DigitalPin.P2, 24, NeoPixelMode.RGB_RGB)
strip.clear()
strip.setBrightness(80)
strip.show()
basic.forever(function () {
    딱 = pins.analogReadPin(AnalogPin.P1)
    if (인피티니스톤 == 21 && 딱 > 200) {
        strip.showRainbow(1, 360)
        while (인피티니스톤 == 21) {
            strip.rotate(1)
            strip.show()
            basic.pause(100)
        }
    }
})
